client/app/tasks/hooks/useTaskDetailsModal.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+import { useRouter, useSearchParams } from 'next/navigation'
+import { TaskTypeWithUser } from '@/_types'
+import { CustomEventType } from '../types'
+
+export const useTaskDetailsModal = (tasks: TaskTypeWithUser[] | undefined) => {
+  const router = useRouter()
+  const searchParams = useSearchParams()
+  const taskId = searchParams.get('taskId')
+
+  const [selectedTask, setSelectedTask] = useState<TaskTypeWithUser | null>(
+    null,
+  )
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  useEffect(() => {
```

Comment:
This useEffect seems to be not working since taskId is always undefined

Reply:
Same 
https://github.com/HAK2024/fairy-share/pull/114#discussion_r1559992137